src/SDL_assert.c

```diff
@@ -160,30 +159,43 @@ SDL_PromptAssertion(const SDL_assert_data *data, void *userdata)
         {   SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT,
                 SDL_ASSERTION_ALWAYS_IGNORE,    "Always Ignore" }
     };
-    char *message;
     int selected;
 
+    char stack_buf[SDL_MAX_LOG_MESSAGE_STACK];
+    char *message = stack_buf;
+    size_t buf_len = sizeof(stack_buf);
+    size_t len;
+
     (void) userdata;  /* unused in default handler. */
 
-    /* !!! FIXME: why is this using SDL_stack_alloc and not just "char message[SDL_MAX_LOG_MESSAGE];" ? */
-    message = SDL_stack_alloc(char, SDL_MAX_LOG_MESSAGE);
-    if (!message) {
-        /* Uh oh, we're in real trouble now... */
-        return SDL_ASSERTION_ABORT;
-    }
-    SDL_snprintf(message, SDL_MAX_LOG_MESSAGE,
-                 "Assertion failure at %s (%s:%d), triggered %u %s:" ENDLINE
-                    "  '%s'",
-                 data->function, data->filename, data->linenum,
-                 data->trigger_count, (data->trigger_count == 1) ? "time" : "times",
-                 data->condition);
+    do {
+        /* Assume the output will fit... */
+        len = SDL_snprintf(message, buf_len,
+            "Assertion failure at %s (%s:%d), triggered %u %s:" ENDLINE "  '%s'",
+                data->function, data->filename, data->linenum,
+                data->trigger_count, (data->trigger_count == 1) ? "time" : "times",
+                data->condition);
+
+        /* .. and if it didn't, allocate a bigger buffer and try again */
+        if (len >= buf_len && message == stack_buf) {
+            buf_len = SDL_MAX_LOG_MESSAGE;
+            message = SDL_malloc(buf_len);
```

Comment:
I see I didn't put the `(char *)` cast on this allocation. I'll fix it if/when there are other revisions requested.